go/cmd/dolt/commands/credcmds/new.go

```diff
@@ -74,7 +76,32 @@ func (cmd NewCmd) Exec(ctx context.Context, commandStr string, args []string, dE
 	help, usage := cli.HelpAndUsagePrinters(commandStr, newShortDesc, newLongDesc, newSynopsis, ap)
 	cli.ParseArgs(ap, args, help)
 
-	_, _, verr := actions.NewCredsFile(dEnv)
+	_, newCreds, verr := actions.NewCredsFile(dEnv)
+
+	if verr != nil {
+		return commands.HandleVErrAndExitCode(verr, usage)
+	}
+
+	err := updateConfigToUseNewCredIfNoExistingCred(dEnv, newCreds)
+	if err != nil {
+		verr = errhand.BuildDError("error: updating user.creds in dolt config to use new credentials").Build()
+		return commands.HandleVErrAndExitCode(verr, usage)
+	} else {
+		return 0
+	}
+}
+
+func updateConfigToUseNewCredIfNoExistingCred(dEnv *env.DoltEnv, dCreds creds.DoltCreds) error {
+	gcfg, hasGCfg := dEnv.Config.GetConfig(env.GlobalConfig)
+
+	if !hasGCfg {
+		panic("global config not found.  Should create it here if this is a thing.")
```

Comment:
if this is a thing?

Reply:
No idea...that was copied from `dolt creds new`